src/04-spin-boson-model.md

```diff
@@ -0,0 +1,800 @@
+        amp = 0
+        for i in range(0,num_states):
+            amp +=  psi0_in_H_basis[i][0][0]*np.exp(-1j*evals[i]*times)*ekets[i][k][0][0]
+        psi[k,:] = amp
+        P[k,:] = amp*np.conj(amp)
+    return P, psi
+```
+
+```python
+times = np.linspace(0.0, 1000.0, 1000)
+P, psi = simulate(H, psi0, times)
+```
+
+Before we make the plots, we must remember to regenerate the state labels.
+
+```python
+bra_labels, ket_labels = make_braket_labels(nm_list)
+```
+
+```python
+plt.figure(figsize=(10,8))
+for i in range(0,P.shape[0]):
+    plt.plot(times, P[i,:], label=f"{ket_labels[i]}")
+plt.ylabel("Probability")
+plt.xlabel("Time")
+plt.legend(loc="right")
+plt.title(f" {H_latex}    ($\Delta E=2.88$, $\omega=1$, $U=0.2$)     (Fig 12)")
+plt.show();
+```
+
+Fig 12 shows us that our ideas were correct - the system starts off with no bosons and, over time, transitions to a state with 3 bosons at the expense of the TSS energy. We can see this energy exchange explicitly by evaluating the expectation values of the various parts of the Hamiltonian.
+
+
+QuTiP does allow us to do this using the [`expect`](http://qutip.org/docs/latest/guide/guide-states.html#expectation-values) function. However, it turns out that we need to create a `Qobj` for every time step in order to use this function and that can be very slow. We will instead directly calculate the expectation value using matrix multiplication, i.e.
+
+
+
+$<H> = \psi^{\dagger}H\psi = \psi^{\dagger} @ (H @\psi) $
+
+Where @ is the matrix multiplication operator and $\dagger$ in this context means taking the complex conjugate.
+
+
+Let's automate this process for all time steps using a function.
+
+```python
+def expectation(operator, states):
+    operator_expect = []
+    for i in range(0,shape(states)[1]):
+        e = np.conj(states[:,i])@ (operator @ states[:,i])
+        operator_expect.append(e)
+    return operator_expect
+```
+
+We can now see how the different parts of the Hamiltonian change over time:
+
+```python
+hamiltonian_expect = expectation(H,psi)
+two_state_expect = expectation(2.88*two_state,psi)
+bosons_expect = expectation(1*bosons,psi)
+interaction_expect = expectation(0.2*interaction,psi)
+```
+
+```python
+plt.figure(figsize=(10,8))
+plt.plot(times, hamiltonian_expect, label="total hamiltonian")
+plt.plot(times, two_state_expect, label="two-state")
+plt.plot(times, bosons_expect, label="bosons")
+plt.plot(times, interaction_expect, label="interaction")
+
+plt.ylabel("Energy")
+plt.xlabel("Time")
+plt.legend(loc="right")
+plt.title(f"Expectation values for {H_latex}    ($\Delta E=2.88$, $\omega=1$, $U=0.2$)     (Fig 13)")
+plt.show();
+```
+
+In Fig 13 we can indeed see the energy exchange that we expect between the TSS and the boson field. We can also see how the interaction energy is moving in sync with the two state energy - in some sense augmenting it. Sometimes people refer to this behaviour as the field is "dressing" the levels of the TSS.
```

Comment:
Very cool! Is the fact that the two state system starts out at a value larger than 1 due to dressing?

Reply:
Not quite. `two state` refers to `\DeltaE/2sigma_z` of the hamiltonian. Since DeltaE = 2.88 then DeltaE/2 =1.44 which is above 1.

Perhaps you looked at the legend and thought it was just $1/2 sigma_z$ from my earlier definition? If so then I'll remake the labels to be clearer

Reply:
![Screenshot 2020-05-18 at 16 17 54](https://user-images.githubusercontent.com/7937044/82230092-2dd41400-9923-11ea-9510-5edddbf796d1.png)
 Now it looks like this - does that make it easier to understand?